boa/src/builtins/number/mod.rs

```diff
@@ -214,6 +214,19 @@ impl Number {
         }
     }
 
+    /// Helper function that formats a float as a ES6-style exponential number string with a given precision.
+    // We can't use the same approach as in `num_to_exponential`
+    // because in cases like (0.999).toExponential(0) the result will be 1e0.
+    // Instead we get the index of 'e', and if the next character is not '-' we insert the plus sign
+    fn num_to_exponential_with_precision(n: f64, prec: usize) -> String {
```

Comment:
```suggestion
    fn to_exponential_with_precision(n: f64, prec: usize) -> String {
```

I'd rename `num_to_exponential` to `f64_to_exponential` and `num_to_exponential_with_precision` to `f64_to_exponential_with_precision`, and extract them as standalone functions. They don't need to be associated functions if they're just used within this module and don't receive a `&self`.

Reply:
Sure, I agree, I just followed what was already in the codebase 😅

Reply:
Actually, what about other helper functions that are in the `impl Number` block? Is there a threshold when a function should be extracted and when it should stay? And would you recommend extracting it out in the same file?

Reply:
If the function is only used internally and it doesn't need a `self` parameter, just put it as a standalone function; you can follow that rule.

We're still cleaning up some parts of the codebase, that's why there's not too much consistency about this 😅

Reply:
Yeah, helper functions can be on the same file